### ISSUE TITLE
Images optimized by C1 is a huge drawback in performance audits

### ISSUE THREAD

**Issue by slacto** [Orckestra/C1-CMS-Foundation#478]
Chrome Audits (by Lighthouse) claims some images can be optimized 90% (!) by using "WebP images" even though I use quality 80 (?q=80).

Please consider rewriting the optimizer using one of the newer image formats such as JPEG-XR and WebP.
http://deanhume.com/home/blogpost/webp---advanced-image-optimization-using-asp-net-mvc/89
https://www.modpagespeed.com/doc/filter-image-optimize

**Comment by DBailey635** [comment]
Backwards compatibilty should be considered when "rewiring" the image optomiser. JPEG is the most widely used image format, and supported in really old browsers. If the team do add WebP or JPEG-XR outputs, the querystring should be used to decide upon output format, eg.: file.jpg?q=60&output=webp

Users would also need to switch from using the `<img>` tags to the `<picture>` tag, ala:

```
<picture>
  <source srcset="img/file.jpg?q=60&output=webp" type="image/webp">
  <source srcset="img/file.jpg?q=60" type="image/jpeg"> 
  <img src="img/file.jpg?q=60" alt="Alt Text!">
</picture>
```

... or use CSS / JavaScript image substitution.

There's a good primer on using WebP images in a website at:

[https://css-tricks.com/using-webp-images/](https://css-tricks.com/using-webp-images/)

**Comment by slacto** [comment]
Yes, and I already switched to< picture> tag in my functions to get image lazy load (https://github.com/verlok/lazyload)

<picture class="topimage" title="@imageTitle">
    <source media="(min-width: 975px)" data-srcset="@Html.C1().MediaUrl(image.Id)?mw=1170&amp;q=80" />
    <source media="(min-width: 751px)" data-srcset="@Html.C1().MediaUrl(image.Id)?mw=974&amp;q=80" />
    <img data-src="@Html.C1().MediaUrl(image.Id)?mw=750&amp;q=80" alt="@imageTitle" width="1170" height="435" />
</picture>

**Comment by slacto** [comment]
This is still an issue
![image](https://user-images.githubusercontent.com/2202838/44657803-6a006c00-a9fe-11e8-9bbd-21e7f8e48dff.png)

**Comment by burningice2866** [comment]
As long as the Image handling and processing isn't plug-able i find this hard to justify since WebP isn't natively supported in the .Net framework and C1 would have to ship various extra dll's just to support this feature.

Its fairly simple though to add own customizations and features to https://github.com/Orckestra/C1-CMS-Foundation/blob/dev/Website/Renderers/ShowMedia.ashx

**Comment by napernik** [comment]
A C1 package for WebP support will come right after the next release

**Comment by napernik** [comment]
The "**Orckestra.Media.ImageFormats.WebP**" package is available online for versions 6.11 and later, source code:

https://github.com/Orckestra/CMS-Packages/commit/d21aef57bd7ac9ac00d86963bcb7c00bb2ca5d2d

Usage:

/media/40ef35a6-2ed9-4daa-a5db-c47ecc13e6d6/Q8djMA/phone.png?mw=300&**mt=image/webp**

Documentation will be provided soon
